gym_xiangqi/envs/xiangqi_env.py

```diff
@@ -1,13 +1,170 @@
 import gym
-# from gym import error, spaces, utils
-# from gym.utils import seeding
+from gym import spaces
+import numpy as np
+
+from gym_xiangqi.xiangqi_game import XiangQiGame
+from gym_xiangqi.piece import Piece, General, Advisor, Elephant
+from gym_xiangqi.piece import Horse, Chariot, Cannon, Soldier
+
+
+PIECE_POINTS = [
+    0.,                     # EMPTY: No point for empty grid
+    float('inf'),           # GENERAL: Priceless for the general
+    2., 2.,                 # ADVISOR: 2.0 points
+    2., 2.,                 # ELEPHANT: 2.0 points
+    4., 4.,                 # HORSE: 4.0 points
+    9., 9.,                 # CHARIOT: 9.0 points
+    4.5, 4.5,               # CANNON: 4.5 points
+    1., 1., 1., 1., 1.,     # SOLDIER: 1.0 point (2.0 after crossing river)
+]
+
+# Piece IDs
+EMPTY = 0
+GENERAL = 1
+ADVISOR_1 = 2
+ADVISOR_2 = 3
+ELEPHANT_1 = 4
+ELEPHANT_2 = 5
+HORSE_1 = 6
+HORSE_2 = 7
+CHARIOT_1 = 8
+CHARIOT_2 = 9
+CANNON_1 = 10
+CANNON_2 = 11
+SOLDIER_1 = 12
+SOLDIER_2 = 13
+SOLDIER_3 = 14
+SOLDIER_4 = 15
+SOLDIER_5 = 16
+
+# Piece Movement Offsets
+ORTHOGONAL = [(-1, 0), (0, 1), (1, 0), (0, -1)]
+DIAGONAL = [(-1, 1), (1, 1), (-1, 1), (-1, -1)]
+ELEPHANT_MOVE = [(-2, 2), (2, 2), (-2, 2), (-2, -2)]
+HORSE_MOVE = [
+    [(-1, 0), (-1, -1)], [(-1, 0), (-1, 1)],
+    [(0, 1), (-1, 1)], [(0, 1), (1, 1)],
+    [(1, 0), (1, 1)], [(1, 0), (1, -1)],
+    [(0, -1), (1, -1)], [(0, -1), (-1, -1)]
+]
```

Comment:
Was the river taken into consideration for these offsets? nvm if it will be considered elsewhere

Reply:
For the elephants and soldiers? Yes it will not allow elephants to cross the river and it will allow soldiers to move side ways after crossing the river.